src/rules/noUnsafeAnyRule.ts

```diff
@@ -40,248 +40,319 @@ export class Rule extends Lint.Rules.TypedRule {
     public static FAILURE_STRING = "Unsafe use of expression of type 'any'.";
 
     public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, undefined, program.getTypeChecker());
+        return this.applyWithWalker(new NoUnsafeAnyWalker(sourceFile, this.ruleName, program.getTypeChecker()));
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>, checker: ts.TypeChecker): void {
-    if (ctx.sourceFile.isDeclarationFile) {
-        // Not possible in a declaration file.
-        return;
+class NoUnsafeAnyWalker extends Lint.AbstractWalker<void> {
+    constructor(sourceFile: ts.SourceFile, ruleName: string, private checker: ts.TypeChecker) {
+        super(sourceFile, ruleName, undefined);
     }
-    return ts.forEachChild(ctx.sourceFile, cb);
 
-    /** @param anyOk If true, this node will be allowed to be of type *any*. (But its children might not.) */
-    function cb(node: ts.Node, anyOk?: boolean): void {
+    public walk(sourceFile: ts.SourceFile) {
+        if (sourceFile.isDeclarationFile) {
+            return; // Not possible in a declaration file.
+        }
+        sourceFile.statements.forEach(this.noCheck);
+    }
+
+    private noCheck = (node: ts.Node) => void this.visitNode(node);
```

Comment:
this method name is a little unintuitive; can you make it more verbose and add a comment explaining its usage?